adoc/admin-stratos-web-console.adoc

```diff
@@ -0,0 +1,233 @@
+objects and must have the Cloud Provider Integration (CPI) enabled. This
+can be achieved for example when deploying {productname} on top of OpenStack.
+
+=== Securing Stratos
+
+It's highly recommended to secure Stratos' web interface using TLS encryption.
+
+This can be done by creating a TLS certificate for Stratos.
+
+== Installation
+
+. Add SUSE helm charts repository
++
+[source,bash]
+----
+helm repo add suse https://kubernetes-charts.suse.com
+----
++
+. Obtain the default `values.yaml` file of the helm chart
++
+[source,bash]
+----
+helm inspect values suse/console > stratos-values.yaml
+----
+
+. Define `admin` user password
++
+Create a secure password for your admin user and write that into the
+`stratos-values.yaml` as value of the `console.localAdminPassword` key.
++
+*Note well:* this step is required to allow the installation of Stratos without
+having any SUSE Cloud Application platform components deployed on the cluster.
+
+. Define the Storage Class to be used
++
+Open the `stratos-values.yaml` file and look for the `storageClass` entry
+defined at the global level, uncomment the line and provide the name of your
+Storage Class.
++
+The values file will have something like that:
++
+[source,yaml]
+----
+# Specify which storage class should be used for PVCs
+storageClass: default
+----
++
+*Note well:* the file has other `storageClass` keys defined inside of some of
+its resources. These can be left empty to rely on the global Storage Class that
+has just been defined.
+
+. Define how to expose Stratos
++
+The web interface of Stratos can be exposed either by using a {kube} `Ingress`
+or a `LoadBalancer` or both of them at the same time. The behaviour is defined
+inside of the `console.service` stanza of the yaml file:
++
+[source,yaml]
+----
+console:
+  service:
+    annotations: []
+    externalIPs: []
+    loadBalancerIP:
+    loadBalancerSourceRanges: []
+    servicePort: 443
+    # nodePort: 30000
+    type: ClusterIP
+    externalName:
+    ingress:
+      ## If true, Ingress will be created
+      enabled: false
+
+      ## Additional annotations
+      annotations: {}
+
+      ## Additional labels
+      extraLabels: {}
+
+      ## Host for the ingress
+      # Defaults to console.[env.Domain] if env.Domain is set and host is not
+      host:
+
+      # Name of secret containing TLS certificate
+      secretName:
+
+      # crt and key for TLS Certificate (this chart will create the secret based on these)
+      tls:
+        crt:
+        key:
+----
+
+. Expose the web UI using a LoadBalancer
++
+The service can be exposes as a `LoadBalancer` one by setting the value of
+`console.service.type` to be `LoadBalancer`.
++
+The `LoadBalancer` resource can be tuned by changing the values of the other
+`loadBalancer*` params specified inside of the `console.service` stanza.
+
+. Expose the web UI using an Ingress
++
+The Ingress resource can be created by setting
+`console.service.ingress.enabled` to be `true`.
++
```

Comment:
Might be worthy to mention ingress.host here. With leaving default blank value leads to
```
Error: render error in "console/templates/ingress.yaml": template: console/templates/__helpers.tpl:151:3:
 executing "ingress.host" at <required "Host name is required" $host>: error calling required: Host name
is required
```

Reply:
Good point, I've added that.